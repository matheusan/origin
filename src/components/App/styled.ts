import styled, { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  html, body {
    margin: 0;
    padding: 0;

    background-color: ${props => props.theme.colors.blueGray10}
  }
`;

const Container = styled.div`
  width: 100%;
`;

export { GlobalStyle, Container };
