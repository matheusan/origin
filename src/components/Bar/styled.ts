import styled from 'styled-components';

export const Bar: JSX.Element = styled.div`
  display: flex;
  top: 0;
  left: 0;
  right: 0;
  height: 80px;

  padding: 24px 56px;
  background-color: ${props => props.theme.colors.neutralWhite};
  align-items: center;

  @media only screen and (max-width: 380px) {
    height: 56px;
  }
`;
