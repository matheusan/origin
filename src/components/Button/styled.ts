import styled from 'styled-components';

export const Button: JSX.Element = styled.button`
  display: block;
  width: ${props => (props.secondary || props.flex ? '100%' : '320px')};
  height: ${props => (props.secondary ? '40px' : '56px')};
  margin: 0 auto;
  border-radius: ${props => (props.secondary ? '8px' : '32px')};
  border: 0;
  outline: 0;
  cursor: pointer;

  background-color: ${props =>
    props.secondary
      ? props.theme.colors.blueGray10
      : props.theme.colors.brandColorPrimary};

  color: ${props =>
    props.secondary
      ? props.theme.colors.brandColorSecondary
      : props.theme.colors.neutralWhite};

  font-family: ${props => props.theme.fonts.primary};
  font-size: 16px;
  line-height: ${props => (props.secondary ? '40px' : '54px')};
  text-align: center;

  &:hover {
    background-color: ${props =>
      props.secondary
        ? props.theme.colors.neutralWhite
        : props.theme.colors.brandColorSecondary};
  }
`;
