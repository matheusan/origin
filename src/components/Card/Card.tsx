import * as React from 'react';

import * as S from './styled';
import findByType from '../../utils/findByType';

const Top = () => null;
const Bottom = () => null;
const Single = () => null;

const Card: React.SFC = ({ children }): JSX.Element => {
  const top = findByType(children, Top);
  const bottom = findByType(children, Bottom);
  const single = findByType(children, Single);

  if (!top && !bottom && !single) {
    return <S.Wrapper>{children}</S.Wrapper>;
  }

  return (
    <S.Wrapper>
      {top && <S.Top>{top.props.children}</S.Top>}
      {bottom && <S.Bottom>{bottom.props.children}</S.Bottom>}
      {single && <S.Single>{single.props.children}</S.Single>}
    </S.Wrapper>
  );
};

Card.Top = Top;
Card.Bottom = Bottom;
Card.Single = Single;

export default Card;
