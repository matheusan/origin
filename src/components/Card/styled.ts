import styled from 'styled-components';

export const Wrapper: JSX.Element = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-items: stretch;
  align-content: stretch;

  margin-top: 24px;
  margin-bottom: 32px;

  border-radius: 8px;
  border: 1px solid ${props => props.theme.colors.blueGray50};
`;

export const Single: JSX.Element = styled.div`
  padding: 16px 28px;

  @media only screen and (max-width: 360px) {
    padding: 12px 32px;
  }
`;

export const Top: JSX.Element = styled.div`
  padding: 24px 32px;

  @media only screen and (max-width: 360px) {
    padding: 24px 27px;
  }
`;
export const Bottom: JSX.Element = styled.div`
  padding: 24px 32px;

  background-color: ${props => props.theme.colors.blueGray10};

  line-height: 16px;
  font-weight: 400;
  font-size: 12px;
  font-family: ${props => props.theme.fonts.primary};
  color: ${props => props.theme.colors.blueGray900};
`;
