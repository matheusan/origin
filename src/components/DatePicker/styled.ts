import styled from 'styled-components';

export const Wrapper: JSX.Element = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: stretch;
  align-content: stretch;

  max-height: 56px;

  border: 1px solid ${props => props.theme.colors.blueGray50};
  background: ${props => props.theme.colors.neutralWhite};
  padding: 3px 5px;
`;

export const Button: JSX.Element = styled.button`
  margin: 0;
  padding: 3px 10px;
  outline: none;
  background-color: transparent;
  border: 0;
  cursor: pointer;

  ${props =>
    props.direction && props.direction === 'right'
      ? `> img { transform: rotate(180deg); }`
      : ''}
`;

export const MonthYear: JSX.Element = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  align-content: stretch;

  margin: 0 10px;
  line-height: 24px;
  font-size: 16px;
  font-family: ${props => props.theme.fonts.primary};
  color: ${props => props.theme.colors.blueGray900};
`;
