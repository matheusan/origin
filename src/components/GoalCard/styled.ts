import styled from 'styled-components';

export const Wrapper: JSX.Element = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: stretch;
  align-content: stretch;

  width: 260px;
  height: 250px;

  padding: 24px;
  margin-bottom: 2em;

  background: ${props => props.theme.colors.neutralWhite};

  border-radius: 8px;
  box-shadow: 0px 16px 32px rgba(30, 42, 50, 0.38);

  @media only screen and (max-width: 380px) {
    margin: 0 auto;
    border-radius: 4px;
    padding: 24px;
  }
`;
