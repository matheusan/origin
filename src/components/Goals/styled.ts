import styled from 'styled-components';

export const Wrapper: JSX.Element = styled.div`
  max-width: 1300px;
  margin: 0 auto;
`;
