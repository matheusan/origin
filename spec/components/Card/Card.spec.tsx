import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Card from '../../../src/components/Card';
import { Top, Bottom, Single } from '../../../src/components/Card/styled';

describe('Card', () => {
  const singleText = 'Single text';
  const topText = 'Top text';
  const bottomText = 'Bottom text';
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Card>{singleText}</Card>);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render two child elements', () => {
    wrapper = shallow(
      <Card>
        <Card.Top>{topText}</Card.Top>
        <Card.Bottom>{bottomText}</Card.Bottom>
      </Card>
    );
    expect(wrapper.find(Top).length).toBe(1);
    expect(wrapper.find(Bottom).length).toBe(1);
  });

  it('should render top child element', () => {
    wrapper = shallow(
      <Card>
        <Card.Top>{topText}</Card.Top>
      </Card>
    );
    expect(wrapper.find(Top).length).toBe(1);
  });

  it('should render bottom child element', () => {
    wrapper = shallow(
      <Card>
        <Card.Bottom>{topText}</Card.Bottom>
      </Card>
    );
    expect(wrapper.find(Bottom).length).toBe(1);
  });

  it('should render single child element', () => {
    wrapper = shallow(
      <Card>
        <Card.Single>{topText}</Card.Single>
      </Card>
    );
    expect(wrapper.find(Single).length).toBe(1);
  });
});
