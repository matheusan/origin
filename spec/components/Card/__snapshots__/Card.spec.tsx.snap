// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders 1`] = `
<styled.div>
  Single text
</styled.div>
`;
